### PR TITLE
Show pending bookings

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- ticket redmine: [https://edu-redmine.sun-asterisk.vn/issues/76148](https://edu-redmine.sun-asterisk.vn/issues/76148)

## WHAT (optional)
- Show all bookings (by statuses)

### CODE REVIEW

**Comment by ghost** [comment]
ready

File: app/models/booking.rb [hunk 1]
```diff
@@ -18,6 +18,31 @@ class Booking < ApplicationRecord
   belongs_to :flight_ticket, optional: true
   has_one :flight, through: :flight_ticket
 
+  scope :ordered_by_status, (lambda do
+    order(Arel.sql("CASE status WHEN 0 THEN 1 WHEN 1 THEN 2 WHEN 2 THEN 3 END"))
+  end)
+  scope :by_tour_name, (lambda do |name|
+    joins(:tour).where("tours.tour_name LIKE ?", "%#{name}%") if name.present?
```
Comment (ghost) [inline app/models/booking.rb]: bọc trong sanitize_sql_like

File: app/models/booking.rb [hunk 2]
```diff
@@ -18,6 +18,31 @@ class Booking < ApplicationRecord
...
+    where("total_price >= ?", total_price.presence ||
+    Settings.booking.search.min_total_price)
+  end)
+  scope :by_started_date, (lambda do |started_date|
+    where("started_date >= ?", started_date.presence ||
+    Settings.booking.search.min_started_date)
+  end)
+  scope :pending, ->{where(status: statuses[:pending])}
+  scope :confirmed, ->{where(status: statuses[:confirmed])}
+  scope :cancelled, ->{where(status: statuses[:cancelled])}
```
Comment (ghost) [inline app/models/booking.rb]: Booking.pending là ra mà đâu phải viết scope đâu

**Review by ghost — COMMENTED** [review]
cmt

**Comment by ghost** [comment]
ready

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -23,6 +21,15 @@ def create
   end
 
   private
+
+  def user
```
Comment (huongnt-2545) [inline app/controllers/users_controller.rb]:
```suggestion
  def load_user
```

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -1,10 +1,8 @@
 class UsersController < ApplicationController
+  before_action :user
```
Comment (huongnt-2545) [inline app/controllers/users_controller.rb]: only actions

**Comment by ghost** [comment]
ready

**Review by ghost — APPROVED** [review]

**Pull request merged**